django/db/models/fields/__init__.py

```diff
@@ -772,6 +769,8 @@ def get_db_prep_lookup(self, lookup_type, value, connection,
             return [self.get_db_prep_value(value, connection=connection,
                                            prepared=prepared)]
         elif lookup_type in ('range', 'in'):
+            if not isinstance(value, (list, tuple)):
```

Comment:
Hmm. I'm not as familiar with this code as you or @akaariai , so it's quite likely that I'm missing something, but this doesn't feel right to me. If the historical contract of `get_db_prep_lookup` is that it takes iterables for `range` and `in` lookups, and it (still) returns iterables for those lookups too, then (given that it is public documented API for custom fields) I think we should maintain that contract, and continue to follow it on the calling side as well, and remove the need for this ugly check. It's just weird for it to now often take single values but then wrap them in a list when they are returned.

Must the `range` and `in` lookups unwrap all iterables every time, even when there is no bilateral transform to apply?

Is unwrapping really needed? Why can't the raw iterable be unwrapped to apply bilateral transforms, but then put back together into a list and passed to `get_db_prep_lookup` just once as a list?

Even if or when unwrapping is needed, why can't the calling code re-wrap each individual item in a length-one list before passing it to `get_db_prep_lookup`? That's the same thing that'll be done here anyway, but it's a lot nicer to do it on the calling side, maintain a simpler and more sensible contract, and get rid of this ugly type check.